UnitsNet/CustomCode/UnitClasses/Density.extra.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright © 2007 by Initial Force AS.  All rights reserved.
+// https://github.com/InitialForce/UnitsNet
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+namespace UnitsNet
+{
+    public partial struct Density
+    {
+        public static Mass operator *(Density density, Volume volume)
+        {
+            return Mass.FromKilograms(density.KilogramsPerCubicMeter * volume.CubicMeters);
+        }
+
+        public static Mass operator *(Volume volume, Density density)
```

Comment:
I was thinking the exact same thing. It was introduced early on and will break compatibility by removing it, so I'm considering making a list of things to do on the next major version bump. 

One advantage over TimeSpan is that it is compatible with our code generation tools as well as operator overloads, but the biggest drawback is duplicate code for both TimeSpan  and Duration cases as well as not understanding why both exists.

It's possible that implementing implicit cast from TimeSpan to Duration could help, since that would remove the need for implementing both cases in tests and operator overloads, but it might not be intuitive that you can pass a TimeSpan to a method that takes Duration.

I'll have to think a little about that one, not sure what is the best approach. Keeping it or removing it.